### PR TITLE
Some improvements

### PR DESCRIPTION
**Pull request by Taowyoo**
- Expose more type to public to enable user to create custom cipher suite
- Bump version of rustls-mbedtls-provider-utils
- Add some utility functions for MbedTlsPkSigningKey
- Expose some utility functions

### CODE REVIEW

File: rustls-mbedtls-provider-utils/src/pk.rs [hunk 1]
```diff
@@ -75,6 +75,21 @@ pub fn rustls_signature_scheme_to_mbedtls_curve_id(signature_scheme: SignatureSc
...
+/// Helper function to get [`rustls::SignatureAlgorithm`] from mbedtls [`mbedtls::pk::Type`]
+pub fn pk_type_to_signature_algo(pk_type: Type) -> rustls::SignatureAlgorithm {
+    use rustls::SignatureAlgorithm;
+    match pk_type {
+        Type::Rsa => SignatureAlgorithm::RSA,
+        Type::Ecdsa => SignatureAlgorithm::ECDSA,
+        Type::RsassaPss => SignatureAlgorithm::RSA,
+        Type::RsaAlt => SignatureAlgorithm::RSA,
+        Type::Eckey => SignatureAlgorithm::ECDSA,
+        Type::EckeyDh => SignatureAlgorithm::Unknown(u8::MAX),
```
Comment (s-arash) [inline rustls-mbedtls-provider-utils/src/pk.rs:87]: Can we have this function return `Option<SignatureAlgorithm>` instead?

File: rustls-mbedcrypto-provider/src/sign.rs [hunk 2]
```diff
@@ -125,47 +138,47 @@ const RSA_SIGNATURE_SCHEME_PREFER_LIST: &[SignatureScheme] = &[
...
 
     fn algorithm(&self) -> rustls::SignatureAlgorithm {
         self.signature_algorithm
     }
 }
 
-fn pk_type_to_signature_algo(pk_type: mbedtls::pk::Type) -> rustls::SignatureAlgorithm {
-    use rustls::SignatureAlgorithm;
+/// Helper function to choose proper [`SignatureScheme`] based on given inputs
+pub fn get_signature_schema_from_offered(
```
Comment (s-arash) [inline rustls-mbedcrypto-provider/src/sign.rs:157]: Do we want this to be public as well?

Comment (Taowyoo) [reply rustls-mbedcrypto-provider/src/sign.rs:157]: Yes, or maybe it's better to put it in `utils` crate?

Comment (s-arash) [reply rustls-mbedcrypto-provider/src/sign.rs:157]: I don't have a preference in that case. I think here is fine.

Comment (s-arash) [reply rustls-mbedcrypto-provider/src/sign.rs:157]: Could you write a doc comment specifying what the function does. (e.g., when it returns `None`, when it expects `ec_signature_scheme` to be `Some(..)`, etc.). Also could you explain the arguments as well?

File: rustls-mbedcrypto-provider/src/sign.rs [hunk 3]
```diff
@@ -125,47 +138,47 @@ const RSA_SIGNATURE_SCHEME_PREFER_LIST: &[SignatureScheme] = &[
...
+            // choose a rsa schema
+            for scheme in rsa_scheme_prefer_order_list {
+                if offered.contains(scheme) {
+                    return Some(*scheme);
+                }
+            }
+            None
+        }
+        mbedtls::pk::Type::Eckey | mbedtls::pk::Type::EckeyDh | mbedtls::pk::Type::Ecdsa => {
+            let scheme = ec_signature_scheme.expect("validated");
```
Comment (s-arash) [inline rustls-mbedcrypto-provider/src/sign.rs:174]: If we need to panic here, could we have a better error message?

File: rustls-mbedtls-provider-utils/src/pk.rs (hunk 1 above)
Comment (Taowyoo) [reply rustls-mbedtls-provider-utils/src/pk.rs:87]: sure

Comment (Taowyoo) [reply rustls-mbedtls-provider-utils/src/pk.rs]: done

File: rustls-mbedcrypto-provider/src/sign.rs (hunk 2 above)
Comment (Taowyoo) [reply rustls-mbedcrypto-provider/src/sign.rs]: Just added. I just moved it to utils crate

File: rustls-mbedcrypto-provider/src/sign.rs (hunk 3 above)
Comment (Taowyoo) [reply rustls-mbedcrypto-provider/src/sign.rs]: updated

**Review by s-arash — APPROVED** [review]

**Pull request merged**